File: algorithms/2104-sum-of-subarray-ranges/sum-of-subarray-ranges.go
package problem2104

// 暴力
func subArrayRanges(nums []int) int64 {
	length := len(nums)
	var result int64 = 0
	for i, m := range nums {
		max := m
		min := m
		for j := i + 1; j < length; j++ {
			if nums[j] > max {
				max = nums[j]
			}
			if nums[j] < min {
				min = nums[j]
			}
			result += int64(max - min)
		}
	}
	return result
}

File: algorithms/2104-sum-of-subarray-ranges/sum-of-subarray-ranges.py
from typing import List


def sub_array_ranges(nums: List[int]) -> int:
    length = len(nums)
    result = 0
    for i in range(length):
        min_num = nums[i]
        max_num = nums[i]
        for j in range(i + 1, length):
            if nums[j] > max_num:
                max_num = nums[j]
            if nums[j] < min_num:
                min_num = nums[j]
            result += max_num - min_num
    return result
